src/node/snapshotter.h

```diff
@@ -201,12 +205,16 @@ namespace ccf
       next_snapshot_indices.push_back(last_snapshot_idx);
     }
 
+    // TODO: At first, it seemed that this could be merged with record_signature
+    // but record_signature is called from a hook and doesn't have a return
+    // value to tell the consensus to create a new ledger chunk...
     bool record_committable(consensus::Index idx)
```

Comment:
I thought this function could trivially be merged with `record_signature()` but it's not since this one returns a `bool` value which indicates whether a new ledger chunk should be created. However, `record_signature()` is called from a generic hook which don't return anything.